### PR TITLE
Ensure that cc ng waits until provided timeout to populate routes

### PR DESCRIPTION
**Pull request by oozie**
Improve DRATs reliability by improving CAPI availability in BBR scripts

Make script require more robust success responses before declaring
server to be "healthy". The number of successes and the timeout are
configurable. 

Signed-off-by: Elena Sharma <esharma@pivotal.io>

Thanks for contributing to the `capi_release`. To speed up the process of reviewing your pull request please provide us with:

* A short explanation of the proposed change:

With this change, the unlock scripts for CC ng job will more reliably signal that CC is back up. 

* An explanation of the use cases your change solves

As a result of this change the subsequent BBR jobs will not hit a 404/502 error when trying to target CC API endpoint.

* Links to any other associated PRs

A similar change had to be committed to UAA:
https://github.com/cloudfoundry/uaa-release/pull/104

* [X] I have viewed signed and have submitted the Contributor License Agreement

* [X] I have made this pull request to the `develop` branch

* [-] I have run CF Acceptance Tests on bosh lite
CATs don't cover this functionality, instead we:
- built the release and deployed it against a deployment
- run a set of DRATs and watched it succeed.

### CODE REVIEW

File: src/capi_utils/monit_utils.sh [hunk 1]
```diff
@@ -37,12 +37,20 @@ function wait_for_server_to_become_unavailable() {
 #
 function wait_for_server_to_become_healthy() {
```
Comment (selzoc) [inline src/capi_utils/monit_utils.sh]: this function is also used in our post-start script, does your change alter that behavior in a significant way?

Comment (oozie) [reply src/capi_utils/monit_utils.sh]: All the semantics of the script have been preserved. The only change in behavior is the condition is satisfied after 10 successful `/v2/info` calls.

Comment (cwlbraa) [reply src/capi_utils/monit_utils.sh]: The default threshold should be one, and y'all should specify 10 for the backup and restore scripts.

**Comment by selzoc** [comment]
Hi @oozie @elenasharma @rowanjacobs (cc @cwlbraa )!  We had to revert this PR as it was causing consistent DRATs failures in our pipeline.  When you originally tested this, was it with more than one API node?  Did you have the change wherein we still advertise the cc route, but return a `503` for nodes that are being `bbr`d?

Here's what's happening:
bbr locks cc1
- cc1 goes into "maintenance mode" where it returns 503
bbr locks cc2
- cc1 goes into "maintenance mode" where it returns 503

At this point, all queries to <API_URL> return a `503` with the message `API down for maintenance` 

 *BBR HAPPENS*

bbr tries to unlock cc1
- cc1 starts
- cc2 is still in maintenance mode
- queries to <API_URL> alternate between a 200 from cc1 and a 503 from cc2, due to the router's round-robin
- therefore, we can never have > 1 consecutive 200 on the cc1 unlock - so the script always fails
- cc1 workers are never started

bbr unlocks cc2
- cc2 starts
- queries to <API_URL> _all_ return 200s, since cc1 and cc2 are both running

Finally, DRATs fails with the message `Error attempting to run post-restore-unlock for job cloud_controller_ng on api/***`

We `set -x` on the bbr scripts so that you can see this happening: https://capi.ci.cf-app.com/teams/main/pipelines/capi/jobs/xena-mysql-drats/builds/1274

Sounds like we need to re-think this approach?

-- Chris && @ericpromislow

**Comment by rizwanreza** [comment]
@oozie We may want to revert these from backports as well?

**Comment by cwlbraa** [comment]
https://pivotal.slack.com/archives/C3LV25ZCM/p1552074407140000

thread has another possible approach

**Comment by ericpromislow** [comment]
Thread's private. Anything in there of general interest?

**Comment by cwlbraa** [comment]
@ericpromislow I was super unclear in the linked thread anyways, rereading it now. But I was trying to suggest that we should verify that the 1 localhost CC responds after a monit start and then verify that (at least one) of them is externally routable.

**Comment by selzoc** [comment]
see https://github.com/cloudfoundry/capi-release/pull/132 for the solution we ended up on

**Pull request merged**